linux/net/rina/ipcps/shim-eth-vlan.c

```diff
@@ -956,31 +956,41 @@ int update_qdisc(struct net_device *    dev,
 	}
 
 	if (string_cmp(dev->qdisc->ops->id,
-		       shim_eth_qdisc_ops.id) == 0)
+		       shim_eth_qdisc_ops.id) == 0) {
+		LOG_INFO("Shim-eth-qdisc already set on this device");
 		return 0;
-
-	sch = qdisc_create_dflt(netdev_get_tx_queue(dev, 0),
-			        &shim_eth_qdisc_ops, 0);
-	if (!sch) {
-		LOG_ERR("Problems creating shim-eth-qdisc");
-		return -1;
-	}
-
-	attr.nla_len = info->qdisc_max_size;
-	attr.nla_type = info->qdisc_enable_size;
-	if (shim_eth_qdisc_init(sch, &attr)) {
-		LOG_ERR("Problems initializing shim-eth-qdisc");
-		qdisc_destroy(sch);
-		return -1;
 	}
 
-	old_qdisc = dev->qdisc;
-
 	if (dev->flags & IFF_UP)
 		dev_deactivate(dev);
 
-	dev_graft_qdisc(netdev_get_tx_queue(dev, 0), sch);
-	dev->qdisc = sch;
+	/* For each TX queue, create a new shim-eth-qdisc instance, attach
+	 * it to the queue and then destroy the old qdisc
+	 */
+	for (q_index = 0; q_index < dev->num_tx_queues; q_index++) {
+		queue = netdev_get_tx_queue(dev, q_index);
+		qdisc = qdisc_create_dflt(queue, &shim_eth_qdisc_ops, 0);
+		if (!qdisc) {
+			LOG_ERR("Problems creating shim-eth-qdisc");
+			continue;
```

Comment:
Ok. Have a look at git commit --amend. Will reduce the number of commits.
On Sep 15, 2015 9:55 AM, Eduard Grasa <notifications@github.com> wrote:In linux/net/rina/ipcps/shim-eth-vlan.c:
>  	if (dev->flags & IFF_UP)
>  		dev_deactivate(dev);
>  
> -	dev_graft_qdisc(netdev_get_tx_queue(dev, 0), sch);
> -	dev->qdisc = sch;
> +	/* For each TX queue, create a new shim-eth-qdisc instance, attach
> +	 * it to the queue and then destroy the old qdisc
> +	 */
> +	for (q_index = 0; q_index < dev->num_tx_queues; q_index++) {
> +		queue = netdev_get_tx_queue(dev, q_index);
> +		qdisc = qdisc_create_dflt(queue, &shim_eth_qdisc_ops, 0);
> +		if (!qdisc) {
> +			LOG_ERR("Problems creating shim-eth-qdisc");
> +			continue;

Right, updated code with a new commit

—Reply to this email directly or view it on GitHub.